### DISCUSSION TITLE
Cannot go to symbol definition in VS

### DISCUSSION THREAD

**Discussion by kostya9** [dotnet/fsharp#11237] [Q&A] [answered]
I am coming from C# background and started playing with F# web apps. 

I have a small Program.fs file I created from the Giraffe Get Started page. I want to see the source code for the `route` function (L9 Program.fs), but when I click `"Go to definition"` in Visual Studio I see `"Cannot navigate to the symbol under the caret."`. I found an issue https://github.com/giraffe-fsharp/Giraffe/issues/426 that says that Giraffe is probably packaged correctly, so I decided to ask here. Is this expected behaviour, an fsharp bug or something is wrong with the Giraffe package?

Program.fs
```f#
open Microsoft.AspNetCore.Hosting
open Microsoft.AspNetCore.Builder
open Giraffe
open Microsoft.Extensions.Hosting
open Microsoft.Extensions.DependencyInjection
open Microsoft.Extensions.Logging

let webApp =
    choose [ route "/ping" >=> text "pong" // I try to go to definition here on the route function
             route "/" >=> htmlFile "/pages/index.html" ]

type Startup() =
    member __.ConfigureServices(services: IServiceCollection) =
        // Register default Giraffe dependencies
        services.AddGiraffe() |> ignore

    member __.Configure (app: IApplicationBuilder) (env: IHostEnvironment) (loggerFactory: ILoggerFactory) =
        // Add Giraffe to the ASP.NET Core pipeline
        app.UseGiraffe webApp


[<EntryPoint>]
let main argv =

    Host
        .CreateDefaultBuilder()
        .ConfigureWebHostDefaults(fun webHostBuilder -> webHostBuilder.UseStartup<Startup>() |> ignore)
        .Build()
        .Run()

    0

```

.fsproj file
```xml
<Project Sdk="Microsoft.NET.Sdk.Web">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net5.0</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <Compile Include="Program.fs" />
  </ItemGroup>

  <ItemGroup>
    <PackageReference Include="Giraffe" Version="4.1.0" />
  </ItemGroup>

</Project>
```

**Comment by cartermp** [comment, marked as answer]
There's a version of this coming in the VS 16.10 update, but only for non-F# constructs. It will enable a decompiled view. For example, this is from `IHostEnvironment`:

![image](https://user-images.githubusercontent.com/6309070/111230184-6f196c80-85a4-11eb-8743-a3c4c9208f94.png)

But as mentioned, this doesn't work yet for an F# project yet.

**Reply by kostya9** [reply]
Thanks! If there is already an issue for this that I can upvote, please let me know.

If somebody else finds this, the feature works in VS Code + Ionide, which is pretty nice.

**Reply by vzarytovskii** [reply]
I don't think we're tracking this anywhere on GH, @TIHan @cartermp correct me if I'm wrong.

**Reply by cartermp** [reply]
The enhancements are tracked here:

https://github.com/dotnet/fsharp/issues/11278
https://github.com/dotnet/fsharp/issues/11279
https://github.com/dotnet/fsharp/issues/11280
